store/sqlstore/user_store.go

```diff
@@ -1316,3 +1316,14 @@ func (us SqlUserStore) ClearAllCustomRoleAssignments() store.StoreChannel {
 		}
 	})
 }
+
+func (us SqlUserStore) InferSystemInstallDate() store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		createAt, err := us.GetReplica().SelectInt("SELECT CreateAt FROM Users ORDER BY CreateAt ASC LIMIT 1")
```

Comment:
If, by chance, there was a user with a`CreateAt` value of `NULL`, this query would pick it. Perhaps write this as `SELECT MIN(CreateAt) FROM Users` instead to ensure we get the first non-`NULL` value?

Reply:
👍: I like your approach of being explicit just to be sure.

Reply:
Done!